microprofile/WFLY-12313_MicroProfile_OpenAPI_1.1.adoc

```diff
@@ -0,0 +1,97 @@
+* Support YAML and JSON for both input (static/pre-generated OpenAPI document) and output formats.
+* MicroProfile OpenAPI TCK must pass
+
+=== Nice-to-Have Requirements
+
+* Optional features of the specification:
+** Multiple deployments:
+*** The specification expects the `/openapi` endpoint to return the logical union of the documented services for all applications in the runtime. See: https://github.com/eclipse/microprofile-open-api/blob/mp-openapi-1.1.1/spec/src/main/asciidoc/microprofile-openapi-spec.adoc#55-multiple-applications
+*** Users might want the ability to disable OpenAPI documentation of a given deployment.
+** HTTPS transport for the `/openapi` endpoint.
+*** If the server/host of a given application defines an HTTPS listener, we will allow it, but not require it.
+*** Additionally, if the Undertow server/host associated with the application does not define an HTTP endpoint (as required by the spec), we should log a WARN message.
+** The MP OpenAPI specification suggests, but does not require, the use of a `format=JSON` or `format=YAML` parameter for the `/openapi` endpoint, used to define content type of the resulting document, in lieu of an `Accept` header.  This is very reasonable, and simple enough to do.  See: https://github.com/eclipse/microprofile-open-api/blob/mp-openapi-1.1.1/spec/src/main/asciidoc/microprofile-openapi-spec.adoc#query-parameters
+* Allow optional authentication/authorization requirements for the `/openapi` endpoint.  I suspect this would be a separately tracked enhancement.
+
+=== Non-Requirements
+
+* The MP OpenAPI specification does not require support for i18n of the OpenAPI document (e.g. via an Accept-Language header), though this would be a reasonable subsequent enhancement.  See: https://github.com/eclipse/microprofile-open-api/blob/mp-openapi-1.1.1/spec/src/main/asciidoc/microprofile-openapi-spec.adoc#71-internationalization
+* The MP OpenAPI specification does not require support for validating the generated OpenAPI model, though this is a reasonable subsequent enhancement. A number of third-party validator exist and could be leveraged for such a feature.  See: https://github.com/eclipse/microprofile-open-api/blob/mp-openapi-1.1.1/spec/src/main/asciidoc/microprofile-openapi-spec.adoc#72-validation
+* The MP OpenAPI specification does not require support for https://www.w3.org/TR/cors/[Cross-Origin Resource Sharing (CORS)].  See: https://github.com/eclipse/microprofile-open-api/blob/mp-openapi-1.1.1/spec/src/main/asciidoc/microprofile-openapi-spec.adoc#73-cross-origin-resource-sharing-cors
+
+== Test Plan
+
+* Basic subsystem unit tests
+* Add the microprofile-tck tests to the MicroProfile TCK testsuite in WildFly.
+* Integration tests should cover vendor specific behavior, not covered by the TCK tests, including:
+** Ensuring that services deployed using a non-root context path are documented with the correct `server` elements.
+** Any multi-deployment or multi-module deployment behavior
```

Comment:
This would be subject to the Multiple deployments "Nice-to-have" requirement, right?

Reply:
Yes.